### ISSUE TITLE
Fatal flaw?

### ISSUE THREAD

**Issue by jfeaver** [code-and-effect/effective_obfuscation#1]
I noticed today that `acts_as_obfuscated` records cannot be reloaded.

In Rails 4.2.1, the reload method is as follows:

``` ruby
    def reload(options = nil)
      clear_aggregation_cache
      clear_association_cache

      fresh_object =
        if options && options[:lock]
          self.class.unscoped { self.class.lock(options[:lock]).find(id) }
        else
          self.class.unscoped { self.class.find(id) }
        end

      @attributes = fresh_object.instance_variable_get('@attributes')
      @new_record = false
      self
    end
```

Given:

User record - id: 1801, to_param: 6894047750

The user record cannot be looked up by its id in the assignment of `fresh_object` above. It raises the exception: `ActiveRecord::RecordNotFound: Couldn't find User with 'id'=2147483647`.

Is this a fatal flaw? Do users of effective_obfuscation need to change their controllers to use finders other than `find()`?

**Comment by matt-riemer** [comment]
It's too bad they don't use `find_by_id` instead of `find` which would trigger the correct behaviour.  Will think about the best approach for this edge case

**Comment by matt-riemer** [comment]
Fixed in 1.0.3
